zeebe/zb-db/src/main/java/io/camunda/zeebe/db/ChecksumProvider.java

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.zeebe.db;
+
+import java.nio.file.Path;
+import java.util.Map;
+
+public interface ChecksumProvider {
+
+  /**
+   * @param snapshotPath path of snapshot to get live file checksums
+   * @return Map containing fileName - checksums pairs, where the checksums are unsigned integers in
+   *     byte array format.
+   */
+  public Map<String, byte[]> getSnapshotChecksums(final Path snapshotPath);
```

Comment:
:x: I think it would be better to have this return `Map<String, Long>`. The conversion from big-endian int to unsigned long is specific to RocksDb and should be encapsulated in the implementation and not leak into the interface.

Reply:
Just stumbled on this, but in the future, we can also use `Object2LongHashMap` from Agrona to also not deal with boxed types ;)